apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.IntegrationTests/BigtableFixture.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright 2017, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax.Grpc;
+using Google.Apis.Auth.OAuth2;
+using Google.Cloud.Bigtable.Admin.V2;
+using Grpc.Core;
+using System;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Google.Cloud.Bigtable.V2.IntegrationTests
+{
+    [CollectionDefinition(nameof(BigtableFixture))]
+    public class BigtableFixture : IDisposable, ICollectionFixture<BigtableFixture>
+    {
+        private const string EmulatorEnvironmentVariable = "BIGTABLE_EMULATOR_HOST";
+        private const string TestProjectEnvironmentVariable = "TEST_PROJECT";
+
+        public const string ColumnFamily1 = "cf1";
+        public const string OtherColumnFamily = "test_data";
+
+        public ProjectName ProjectName { get; private set; }
+        public InstanceName DefaultInstanceName { get; private set; }
+        public TableName DefaultTableName { get; private set; }
+
+        public BigtableInstanceAdminClient DefaultInstanceAdminClient { get; private set; }
+        public BigtableTableAdminClient DefaultTableAdminClient { get; private set; }
+        public BigtableClient DefaultTableClient { get; private set; }
+
+        public Channel EmulatorChannel { get; }
+
+        public BigtableFixture()
+        {
+            string emulatorHost = Environment.GetEnvironmentVariable(EmulatorEnvironmentVariable);
+
+            string projectId;
+            if (!string.IsNullOrEmpty(emulatorHost))
+            {
+                projectId = "emulator-test-project";
+                EmulatorChannel = new Channel(emulatorHost, ChannelCredentials.Insecure);
+            }
+            else
+            {
+                projectId = Environment.GetEnvironmentVariable(TestProjectEnvironmentVariable);
+                if (string.IsNullOrEmpty(projectId))
+                {
+                    throw new InvalidOperationException(
+                        $"Please set either the {EmulatorEnvironmentVariable} or {TestProjectEnvironmentVariable} environment variable before running tests");
+                }
+            }
+
+            ProjectName = new ProjectName(projectId);
+
+            Task.Run(InitBigtableInstanceAndTable).Wait();
+        }
+
+        private async Task InitBigtableInstanceAndTable()
+        {
+            DefaultInstanceName = new InstanceName(ProjectName.ProjectId, "i" + Guid.NewGuid().ToString("N"));
```

Comment:
If we're not using the emulator, we won't be creating the instance as far as I can see. How does that work?

Reply:
With the emulator, you can just interact with any instance name. Actually, I'm not sure if the instance name is simply ignored or if it keeps separate tables per instance name used. I'll test this out.

Reply:
Oh sorry, misread your question. If we're not using the emulator (there is no channel for it), we DO create the instance. If we're using the emulator, we don't.

Reply:
Ah, with you - yes, I'd misread the condition.